### PR TITLE
[OpenSRP 2] Fix sync failure - Regression

### PR DESCRIPTION
**Pull request by f-odhiambo**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by pld** [comment]
awesome! can you please add a test that fails on main and passes with these changes?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt [hunk 1]
```diff
@@ -216,12 +218,12 @@ class RegisterFragment : Fragment(), OnSyncListener {
         // Show error message in snackBar message
         // syncJobStatus.exceptions may be null when worker fails; hence the null safety usage
         val hasAuthError =
-          syncJobStatus.exceptions.any {
+          syncJobStatus.exceptions?.any {
             it.exception is HttpException && (it.exception as HttpException).code() == 401
           }
-        Timber.e(syncJobStatus.exceptions.joinToString { it.exception.message ?: "" })
+        Timber.e(syncJobStatus?.exceptions?.joinToString { it.exception.message ?: "" })
```
Comment (ndegwamartin) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:224]: I think we might need to revert these, it appears the properties are already null safe cc @ellykits

Comment (pld) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:224]: yes agree and above

Comment (pld) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:224]: and below those changes aren't necessary

**Pull request merged**